assets/src/hooks/useDetour.ts

```diff
@@ -28,52 +27,21 @@ export enum DetourState {
   Finished,
 }
 
-const selectFinishedState = (
-  snapshot: SnapshotFrom<typeof createDetourMachine>
-) => {
-  /*
-   * There's probably a better way to do this? Tags or maybe context?
-   * Tags seem more appropriate, but weird to manage Out-Of-Bounds state via tags.
-   * Maybe this entire API call could be moved to and managed by an actor
-   * combined with parallel child states within the state machine?
-   * -- https://stately.ai/docs/promise-actors */
-  const isFinishedDrawing = snapshot.matches({
-    "Detour Drawing": { Editing: "Finished Drawing" },
-  })
-  const isSharingDetour = snapshot.matches({ "Detour Drawing": "Share Detour" })
-  const isInFinishedDetourState = isFinishedDrawing || isSharingDetour
-  return isInFinishedDetourState
-}
-
-/** This selects the detour waypoints in-between the detour start and end points  */
-const selectWaypoints = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
-  snapshot.matches({
-    "Detour Drawing": { Editing: "Finished Drawing" },
-  })
-    ? snapshot.context.waypoints.slice(1, -1)
-    : snapshot.matches({ "Detour Drawing": { Editing: "Place Waypoint" } })
-    ? snapshot.context.waypoints.slice(1)
-    : []
-
-const selectStartPoint = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
-  snapshot.context.waypoints.at(0)
-
-const selectEndPoint = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
-  (selectFinishedState(snapshot) && snapshot.context.waypoints.at(-1)) ||
-  undefined
-
 export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
-  const [snapshot, send, createDetourActor] = useMachine(createDetourMachine)
-  const startPoint = useSelector(createDetourActor, selectStartPoint)
-  const endPoint = useSelector(createDetourActor, selectEndPoint)
+  const [snapshot, send] = useMachine(createDetourMachine)
+  const startPoint = snapshot.context.startPoint
+  const endPoint = snapshot.context.endPoint
```

Comment:
```suggestion
  const { startPoint, endPoint } = snapshot.context
```

Reply:
merge `waypoints` into this too

Reply:
[Ah yes - perfect](https://github.com/mbta/skate/pull/2629/commits/13cd7a4011b8f95630b1feef672047f576c4a3c1)